modules/tinymce/src/core/main/ts/mode/Readonly.ts

```diff
@@ -156,14 +176,33 @@ const processReadonlyEvents = (editor: Editor, e: Event): void => {
 };
 
 const registerReadOnlySelectionBlockers = (editor: Editor): void => {
+  editor.on('beforeinput paste', (e) => {
+    if (isReadOnly(editor) && editor.mode.isSelectionEnabled()) {
```

Comment:
I feel like with these new blockers it doesn't matter if selection is enabled. They should always block in readonly mode.

```suggestion
    if (isReadOnly(editor)) {
```